### DISCUSSION TITLE
iOS Back Handler

### DISCUSSION THREAD

**Discussion by vk-login-auth** [adrielcafe/voyager#226] [Ideas]
https://github.com/adrielcafe/voyager/assets/55690452/9e194c58-840a-4374-87e7-84158c3540b8

I can send a code of how I implemented this feature using Voyager as a core navigator.

**Comment by Syer10** [comment]
Is this the standard for iOS apps? It looks very nice

**Reply by vk-login-auth** [reply]
Yes, swiping right to close any menu is a standard feature for iOS apps. This feature was introduced in iOS 11, which was released in 2017.

Apple's Human Interface Guidelines state that "swiping right to dismiss a modal view is a common gesture that users expect to work." This includes menus, sheets, and other modal views.

There are a few exceptions to this rule. For example, the Control Center and Notification Center cannot be closed with a swipe gesture. However, most other menus and modal views can be closed by swiping right.

This gesture is convenient for users because it allows them to quickly close menus and modal views without having to tap a button. It is also consistent with other iOS gestures, such as swiping left to go back to the previous screen.

Here are some examples of iOS apps that use the swipe right to close menu gesture:

Settings
Mail
Messages
Photos
Music
Safari
Notes
Calendars
Reminders
Maps
Some third-party iOS apps also use this gesture, but not all of them do. @Syer10

**Reply by vk-login-auth** [reply]
Here is the [link](https://medium.com/@manassehkola/swipe-to-reveal-previous-screen-state-in-jetpack-compose-e2f3226d454a) for Swipe to reveal the previous screen. We have to add this feature to the core Navigator (voyager) and animate using LocalNavigatorStateHolder.

**Comment by DevSrSouza** [comment]
@vk-login-auth can you share your code to help us bring this feature to Voyager?

**Comment by oralbeks** [comment]
@vk-login-auth Great job!

**Comment by DevSrSouza** [comment]
I have implemented the Swipe Back extension for Voyager, you can checkout here.
https://github.com/adrielcafe/voyager/issues/144#issuecomment-1763186873

I'm planning in refining this implementation soon, but, this unblocks any one that wants this in iOS.
